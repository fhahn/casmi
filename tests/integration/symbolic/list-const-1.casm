function (symbolic) a: -> List(Int)
function (symbolic) b: -> List(Int)

init main

rule main = {
  a := [1,2,3]
  b := []
  program(self) := undef
}
