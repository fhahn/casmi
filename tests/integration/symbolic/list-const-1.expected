forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,fcons(eEmptyList,1,sym2)).
tff(symbolNext, type, sym3: $int).
fof(id1,hypothesis,fcons(sym2,2,sym3)).
tff(symbolNext, type, sym4: $int).
fof(id2,hypothesis,fcons(sym3,3,sym4)).
tff(symbolNext, type, sym5: $int).
fof(id3,hypothesis,sta(1,sym5)).%CREATE: a
tff(symbolNext, type, sym6: $int).
fof(id4,hypothesis,stb(1,sym6)).%CREATE: b
fof(id5,hypothesis,sta(2,sym4)).%UPDATE: a
fof(id6,hypothesis,stb(2,eListEmpty)).%UPDATE: b
fof(final0,hypothesis,sta(0,sym4)).%FINAL: a
fof(final1,hypothesis,stb(0,eListEmpty)).%FINAL: b

